pkg/sql/sem/tree/type_check.go

```diff
@@ -2175,16 +2175,27 @@ func typeCheckSubqueryWithIn(left, right *types.T) error {
 func typeCheckComparisonOp(
 	ctx context.Context, semaCtx *SemaContext, op treecmp.ComparisonOperator, left, right Expr,
 ) (_ TypedExpr, _ TypedExpr, _ *CmpOp, alwaysNull bool, _ error) {
+	// If we have a ParenExpr, unwrap it right away - we would do it anyway in
+	// ParenExpr.TypeCheck, but it could be too late for other logic in this
+	// method to kick in.
+	if p, ok := left.(*ParenExpr); ok {
```

Comment:
consider using `tree.StripParens` here -- it will recursively unwrap the parentheses